### DISCUSSION TITLE
Week 7 - Debugging

### DISCUSSION THREAD

**Discussion by derekscholes** [USU-ITLS-x265-23-summer/class-repo#138] [General]
# Debugging 
- I believe that my basic understanding of JavaScript is still insufficient to be able to apply it well in a website. I feel like I understand how things are written and how to debug some small issues but I don't have practice adding it to HTML which totally confuses me. I will definitely be using debugging techniques a lot in implementing JavaScript. I will most likely be using console.log() as well as verifying order of actions within the script. One thing I also like is being "concise" which was discussed in the video and in some of FCC. For me this will allow me to be more organized and help me understand what I am coding.
- Anti-pattern: One thing I will attempt to do is avoid overuse of variables in the global space as well as variables that are simple that don't define their function. Hopefully this will avoid issues of using incorrect variables as well as helping understand the purpose of the function.

# Project Task
- I feel like sticking with just a CV isn't a challenge. I will soon finish my personal statement and CV which I will post to my site. I also have a contact form in there. Maybe I was thinking of adding presentations I created and other documents from school. In addition, I will be attempting to use javascript to make collapsible tabs as well as incorporate videos or interactive files. I am just unsure how to get started on those things. **Does anyone think I should incorporate anything else into my site? Or is this too much?**
- Here is a current list of tasks I have:
  - Finish personal statement and CV
  - Add personal statement and CV to web page
  - Create CSS for each page that changes accordingly to different devices (and organized by page)
  - Create javascript for heading on each page to have a collapsable tab 
  - If my CV is long enough, I will create a Table of Contents with links to specific portions of the CV web page to allow auto scroll to that portion.
  - If I have time, I will add previously written articles to my blog section

**Comment by coopsand** [comment]
I think that you have a great set of tasks for your final project!

**Comment by metasean** [comment]
> I believe that my basic understanding of JavaScript is still insufficient to be able to apply it well in a website.

You might want to review:
- https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_JavaScript_within_a_webpage
- https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/JavaScript_basics
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script
